### PR TITLE
MES-3174: Adding the extra fields

### PR DESCRIPTION
**Pull request by szabikr**
# Description and relevant Jira numbers

Mapping the Rekey properties to the RSIS fields

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: src/functions/uploadToRSIS/application/data-mapper/common-mapper.ts [hunk 1]
```diff
@@ -38,7 +39,7 @@ export const mapCommonData = (result: ResultUpload): DataField[] => {
     field('DATE_OF_TEST', testDateTime.format('YYMMDD')),
     field('TIME', testDateTime.format('HHmm')),
     field('DTC_AUTHORITY_CODE', r.journalData.testCentre.costCode),
-    field('STAFF_NO', Number(r.journalData.examiner.staffNumber).toString()), // get rid of any leading zeros, if any
+    // field('STAFF_NO', r.examinerConducted.toString()), // get rid of any leading zeros, if any
```
Comment (chrisnappin) [inline src/functions/uploadToRSIS/application/data-mapper/common-mapper.ts]: delete this line?

File: src/functions/uploadToRSIS/application/data-mapper/__tests__/common-mapper.spec.ts [hunk 2]
```diff
@@ -551,6 +557,33 @@ describe('mapCommonData', () => {
...
+
+  it('Should reject a test result with missing mandatory data (keyed staff num)', () => {
+    const missingMandatory = cloneDeep(minimalInput);
+    if (missingMandatory.testResult) {
+      delete missingMandatory.testResult.examinerKeyed;
+    }
+    expect(() => mapCommonData(missingMandatory))
+      .toThrow(new MissingTestResultDataError('examinerKeyed'));
+  });
+
```
Comment (chrisnappin) [inline src/functions/uploadToRSIS/application/data-mapper/__tests__/common-mapper.spec.ts:667]: Liking the extra tests. Could you please add/update tests to include all of the various other rekey fields?

**Review by chrisnappin — COMMENTED** [review]
Couple of comments added

File: src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { get } from 'lodash';
+import { get, isEmpty } from 'lodash';
```
Comment (buzzub) [inline src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts]: don't need `isEmpty` anymore

**Review by buzzub — APPROVED** [review]
Just one thing to remove

**Review by MarkPatt89 — APPROVED** [review]

File: src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts (hunk 3 above)
Comment (szabikr) [reply src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts]: removed

File: src/functions/uploadToRSIS/application/data-mapper/common-mapper.ts (hunk 1 above)
Comment (szabikr) [reply src/functions/uploadToRSIS/application/data-mapper/common-mapper.ts]: removed

File: src/functions/uploadToRSIS/application/data-mapper/__tests__/common-mapper.spec.ts (hunk 2 above)
Comment (szabikr) [reply src/functions/uploadToRSIS/application/data-mapper/__tests__/common-mapper.spec.ts:667]: Added an extra test where I check all the additional rekey reason fields as well.

**Review by ammarhaiderbjss — APPROVED** [review]
😍

**Review by chrisnappin — APPROVED** [review]
Cheers Szabi

**Pull request merged**